File: nexus-inventory-backend/src/main/java/com/nexus/inventory/dto/ProductDTO.java
package com.nexus.inventory.dto;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductDTO {
    private Long id;

    @NotBlank(message = "SKU is required")
    private String sku;

    @NotBlank(message = "Product name is required")
    private String name;

    @NotNull(message = "Price is required")
    @Min(value = 0, message = "Price must be >= 0")
    private BigDecimal price;

    @NotNull(message = "Stock is required")
    @Min(value = 0, message = "Stock must be >= 0")
    private Integer stock;

    @NotNull(message = "Minimum stock level is required")
    @Min(value = 0, message = "Min stock level must be >= 0")
    private Integer minStockLevel;

    private Long categoryId;
    private String categoryName;
    private String stockStatus;
}

File: nexus-inventory-mobile/app/src/main/java/com/nexus/inventory/data/dto/ProductDto.kt
package com.nexus.inventory.data.dto

import com.google.gson.annotations.SerializedName
import com.nexus.inventory.domain.model.Product
import com.nexus.inventory.domain.model.StockStatus
import java.math.BigDecimal

data class ProductDto(
    @SerializedName("id") val id: Long,
    @SerializedName("sku") val sku: String,
    @SerializedName("name") val name: String,
    @SerializedName("price") val price: BigDecimal,
    @SerializedName("stock") val stock: Int,
    @SerializedName("minStockLevel") val minStockLevel: Int,
    @SerializedName("categoryId") val categoryId: Long?,
    @SerializedName("categoryName") val categoryName: String?,
    @SerializedName("stockStatus") val stockStatus: String?
)

fun ProductDto.toDomain(): Product {
    val status = when (stockStatus) {
        "OUT_OF_STOCK" -> StockStatus.OUT_OF_STOCK
        "LOW" -> StockStatus.LOW
        else -> if (stock == 0) StockStatus.OUT_OF_STOCK else if (stock <= minStockLevel) StockStatus.LOW else StockStatus.SUFFICIENT
    }

    return Product(
        id = id,
        sku = sku,
        name = name,
        price = price,
        stock = stock,
        minStockLevel = minStockLevel,
        categoryId = categoryId,
        categoryName = categoryName ?: "Uncategorized",
        stockStatus = status
    )
}
